lambda-notes/src/App.js

```diff
@@ -34,115 +40,74 @@ class App extends Component {
-          },
-  
-      ]
+    // Run this to initialize the database only runs once ever!!
+    // this.initializeFireBaseData()
+
+    //Listen for changes to any value in the FB RT DB reference
+    notesRef.on('value', (snapshot) => {
+
+      if (snapshot.val){
+        // Since firebase stores object, we need to convert it to an array of object to work with our architecture
+        let notes = Object.entries(snapshot.val()).map( entry => entry[1])
+
+        //Set state shorthand, when the key name is the same as the name of the value variable
+        console.log(notes)
+        this.setState({notes})  
+      }
+
+    });
+  }
+
+  initializeFireBaseData = () => {
+    initialDataArr.forEach(obj => {
+      let noteKey = notesRef.push() // Blank slot waitin to be filled
+      console.log(noteKey)
+      noteKey.set({ // set the blank slot 
+        id: noteKey.key, // access the key of the slot
+        title:obj.title,
+        tags:obj.tags,
+        textBody:obj.textBody
+      })
     })
   }
 
   saveNewNote = (noteTitle, noteTextBody) => {    
+    let newNoteInFB = notesRef.push()
+
     let newNote = {
-      "id":Date.now(),
+      "id": newNoteInFB.key,
       "tags": ["tag", "doctor4"],
       "title": noteTitle,
       "textBody":noteTextBody
     }
+    
+    newNoteInFB.set(newNote)
 
-    let notes = [...this.state.notes, newNote];
-
-    this.setState({notes})
   }
 
   editNote = (id, title, textBody) => {
-    // Make a copy of this.state.notes
-    let notes = [...this.state.notes]
-    
-    //Grab the index of note we are editing
-    let indexOfEditNoteId = notes.findIndex((obj => obj.id == id))
-    
-    // Update the contents
-    notes[indexOfEditNoteId].title = title;
-    notes[indexOfEditNoteId].textBody = textBody;
-    
-    // Update State
-    this.setState({notes})
+    //Reference the id and then call update on the field, via shorthand
+    notesRef.child(id).update({title})
+    notesRef.child(id).update({textBody})
   }
 
   onDeleteLinkClick = (id) => {
     this.setState({
       deleteNote: {
         deleting:true,
-        id:Number(id),
+        id:id,
       }
     })
   }
 
   deleteNoteClick = () => {
-    let notes = [...this.state.notes].filter( note => note.id !== this.state.deleteNote.id)
-    this.setState({notes}, this.deletingCompleted)
+    // let notes = [...this.state.notes].filter( note => note.id !== this.state.deleteNote.id)
```

Comment:
Still needed? If still so, can probably use `const` here. `const` with objects and arrays work really well... just don't change the data type! 😄 The content can change though with no problem for the most part, or adding new key/value pairs 👍